### PR TITLE
✨ Source Square - Add Bank Account stream

### LINKED ISSUE #16856: Source Square: new stream Bank Accounts
## Tell us about the problem you're trying to solve
Returns a list of [BankAccount](https://developer.squareup.com/reference/square/objects/BankAccount) objects linked to a Square account.

## Describe the solution you’d like
Research pros and cons for implemeting this stream.
For this stream BANK_ACCOUNTS_READ scope is obligatory

### PR DESCRIPTION
**Pull request by avirajsingh7**
Closes #16856 

Added [Bank Account](https://developer.squareup.com/reference/square/objects/BankAccount) stream in Source Square

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks for the contribution 🚀 . I added it to the team backlog and we're planning to start next week the review process.

File: airbyte-integrations/connectors/source-square/source_square/manifest.yaml [hunk 1]
```diff
@@ -289,6 +289,13 @@ definitions:
         request_body_json:
           limit: "{{ 500 }}"
 
+  banks_accounts_stream:
+    $ref: "#/definitions/base_stream"
+    $parameters:
+      name: "banks_accounts"
+      primary_key: "id"
+      path: "/bank-accounts"
```
Comment (archangelic) [inline airbyte-integrations/connectors/source-square/source_square/manifest.yaml:297]:
```suggestion
  bank_accounts_stream:
    $ref: "#/definitions/base_stream"
    $parameters:
      name: "bank_accounts"
      primary_key: "id"
      path: "/bank-accounts"
```

File: airbyte-integrations/connectors/source-square/source_square/manifest.yaml [hunk 2]
```diff
@@ -303,6 +310,7 @@ streams:
   - "#/definitions/refunds_stream"
   - "#/definitions/payments_stream"
   - "#/definitions/orders_stream"
+  - "#/definitions/banks_accounts_stream"
```
Comment (archangelic) [inline airbyte-integrations/connectors/source-square/source_square/manifest.yaml:313]:
```suggestion
  - "#/definitions/bank_accounts_stream"
```

**Review by archangelic — CHANGES_REQUESTED** [review]
Thank you so much for your contribution. I have submitted some changes below. Additionally, can you update `Dockerfile` and `metadata.yaml` to version `1.2.0`?

**Comment by archangelic** [comment]
@avirajsingh7 Are you able to make the suggested changes? I can't seem to push changes to your branch myself.

**Comment by avirajsingh7** [comment]
@archangelic done changes

**Review by archangelic — APPROVED** [review]

**Comment by archangelic** [comment]
Congrats! I merged it in another PR to fix some things!

**Pull request closed without merging**